aws/function.ts

```diff
@@ -31,23 +31,30 @@ export class Function extends pulumi.ComponentResource {
                 opts?: pulumi.ResourceOptions) {
         super("cloud:function:Function", name, { handler: handler }, opts);
 
-        // First allocate a function.
-        const options: aws.serverless.FunctionOptions = {
-            policies: [...getComputeIAMRolePolicies()],
-            memorySize: functionMemorySize,
-            includePaths: functionIncludePaths,
-            includePackages: functionIncludePackages,
-        };
+        const policies = [...getComputeIAMRolePolicies()];
+        let vpcConfig: aws.serverless.FunctionOptions["vpcConfig"] | undefined;
+
         if (runLambdaInVPC) {
             const network = getOrCreateNetwork();
             // TODO[terraform-providers/terraform-provider-aws#1507]: Updates which cause existing Lambdas to need to
             //     add VPC access will currently fail due to an issue in the Terraform provider.
-            options.policies!.push(aws.iam.AWSLambdaVPCAccessExecutionRole);
-            options.vpcConfig = {
+            policies.push(aws.iam.AWSLambdaVPCAccessExecutionRole);
+            vpcConfig = {
                 securityGroupIds: pulumi.all(network.securityGroupIds),
                 subnetIds: pulumi.all(network.subnetIds),
             };
         }
+
+        // First allocate a function.
+        const options: aws.serverless.FunctionOptions = {
+            policies,
+            vpcConfig,
+            memorySize: functionMemorySize,
+            includePaths: functionIncludePaths,
+            includePackages: functionIncludePackages,
+        };
```

Comment:
This changed because the fields of FunctionOptions became readonly.  the old code made the object and then mutated it.  Technically readonly is just a TS'ism, and i oculd have worked around it.  But this was just easier to reorder to make the object last, once the props were available.